airbyte-webapp/src/components/hooks/services/useDbMigrationHook.tsx

```diff
@@ -0,0 +1,32 @@
+import DbMigrationResource, {
+  DbMigrationInfoResult,
+  DbMigrationMigrateResult,
+} from "core/resources/DbMigration";
+import { useAnalytics } from "../useAnalytics";
+import { useFetcher } from "rest-hooks";
+
+type DbMigrationService = {
+  info: (database: string) => Promise<DbMigrationInfoResult>;
+  migrate: (database: string) => Promise<DbMigrationMigrateResult>;
+};
+
+const useDbMigration = (): DbMigrationService => {
+  const analyticsService = useAnalytics();
+  const getMigrationInfo = useFetcher(DbMigrationResource.infoShape());
+  const runMigration = useFetcher(DbMigrationResource.migrateShape());
+
+  const info = async (database: string) => {
```

Comment:
its better to wrap this methods in `useCallback` and wrap result into `useMemo` as this is supposed to be a service